internal/provider/workflow_job_resource.go

```diff
@@ -0,0 +1,320 @@
+	var data WorkflowJobResourceModel
+
+	// Read Terraform plan data into workflow job resource model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Create new Workflow Job from workflow job template
+	resp.Diagnostics.Append(r.LaunchWorkflowJob(&data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+func (r WorkflowJobResource) Delete(_ context.Context, _ resource.DeleteRequest, _ *resource.DeleteResponse) {
+}
+
+// CreateRequestBody creates a JSON encoded request body from the workflow job resource data
+func (r *WorkflowJobResourceModel) CreateRequestBody() ([]byte, diag.Diagnostics) {
+	var diags diag.Diagnostics
+	var inventoryID int64
+
+	// Use default inventory if not provided
+	if r.InventoryID.ValueInt64() == 0 {
+		inventoryID = 1
+	} else {
+		inventoryID = r.InventoryID.ValueInt64()
+	}
+
+	// Convert workflow job resource data to API data model
+	workflowJob := WorkflowJobAPIModel{
+		ExtraVars: r.ExtraVars.ValueString(),
+		Inventory: inventoryID,
+	}
+
+	// Create JSON encoded request body
+	jsonBody, err := json.Marshal(workflowJob)
+	if err != nil {
+		diags.AddError(
+			"Error marshaling request body",
+			fmt.Sprintf("Could not create request body for workflow job resource, unexpected error: %s", err.Error()),
+		)
+		return nil, diags
+	}
+	return jsonBody, diags
+}
+
+// ParseHttpResponse updates the workflow job resource data from an AAP API response
+func (r *WorkflowJobResourceModel) ParseHttpResponse(body []byte) diag.Diagnostics {
+	var diags diag.Diagnostics
+
+	// Unmarshal the JSON response
+	var resultApiWorkflowJob WorkflowJobAPIModel
+	err := json.Unmarshal(body, &resultApiWorkflowJob)
+	if err != nil {
+		diags.AddError("Error parsing JSON response from AAP", err.Error())
+		return diags
+	}
+
+	// Map response to the job resource schema and update attribute values
+	r.Type = types.StringValue(resultApiWorkflowJob.Type)
+	r.URL = types.StringValue(resultApiWorkflowJob.URL)
+	r.Status = types.StringValue(resultApiWorkflowJob.Status)
+	r.TemplateID = types.Int64Value(resultApiWorkflowJob.TemplateID)
+	r.InventoryID = types.Int64Value(resultApiWorkflowJob.Inventory)
+	diags = r.ParseIgnoredFields(resultApiWorkflowJob.IgnoredFields)
+	return diags
+}
+
+func (r *WorkflowJobResourceModel) ParseIgnoredFields(ignoredFields map[string]interface{}) (diags diag.Diagnostics) {
+	r.IgnoredFields = types.ListNull(types.StringType)
+	var keysList = []attr.Value{}
+
+	for k := range ignoredFields {
+		key := k
+		if v, ok := keyMapping[k]; ok {
+			key = v
+		}
+		keysList = append(keysList, types.StringValue(key))
+	}
+
+	if len(keysList) > 0 {
+		r.IgnoredFields, _ = types.ListValue(types.StringType, keysList)
```

Comment:
Why ignore the `diag.Diagnostics` return value and always use the default return variable?